In [5]:
import researchpy
import pandas as pd
from pandas import read_csv


# Load the phone dataset

In [6]:
# Open Dataset
df = read_csv(r"..\Datasets\PhonePreferences.csv")
df

Sex Phone preference
0      Male           Iphone
1      Male           Iphone
2      Male           Iphone
3      Male           Iphone
4      Male           Iphone
..      ...              ...
349  Female          Samsung
350  Female          Samsung
351  Female          Samsung
352  Female          Samsung
353  Female          Samsung

[354 rows x 2 columns]

# Frequency table

In this case there are only two variables, however even if there were more you should select two variables you want to compare.

In [7]:
var1 = df['Phone preference']
var2 = df['Sex']

#Get the count/frequency 
contTable = pd.crosstab(var1, var2)
contTable

Sex               Female  Male
Phone preference              
Iphone                34    61
Other                 71    35
Samsung               88    65

# Run Chi-Square 

In [10]:
crosstab, results = researchpy.crosstab(var1, var2, test='chi-square')
display(crosstab)
display(results)

Sex          
Sex              Female Male  All
Phone preference                 
Iphone               34   61   95
Other                71   35  106
Samsung              88   65  153
All                 193  161  354

Chi-square test  results
0  Pearson Chi-square ( 2.0) =   20.6336
1                    p-value =    0.0000
2                 Cramer's V =    0.2414

# Calculate degree of freedom

We can see that Cramér's V = 0.2414, to identify the degree of the results we need to fiend the degree of freedom
dof = (r-1)(c-1)

In [11]:

dof = (contTable.shape[0]-1)*(contTable.shape[1]-1)
dof

2

dof = 2, according to the lookup table, a 0.2414 will have a medium association.